implementations/rust/ockam/ockam_identity/src/secure_channel/handshake/handshake_worker.rs

```diff
@@ -133,7 +133,18 @@ impl Worker for HandshakeWorker {
 
     async fn shutdown(&mut self, _context: &mut Self::Context) -> Result<()> {
         if let Some(handler) = &self.decryptor_handler {
-            handler.shutdown().await?
+            let encryptor_address = &handler.addresses.encryptor;
+            if let Some(entry) = self.secure_channels.secure_channel_registry.unregister_channel(encryptor_address) {
+                let err = _context.stop_worker(entry.decryptor_messaging_address().clone()).await.err();
```

Comment:
It's probably very unclear from the public API, but the messaging address corresponds to the same Decryptor worker, so we don't need to shut it down here. Shutting down any of the Worker's Addresses leads to shutting down the whole Worker.